provisioning/README.md

```diff
@@ -0,0 +1,109 @@
+taken this secure element, paired it with a nice Cortex-M microcontroller on a
+board to make a device. Now, you'd like to be able to use this key from Mbed
+Crypto. How does Mbed Crypto know the key exists? We can tell Mbed Crypto by
+running a provisionining application, which is typically separate from the
+primary application and may be run as a final device assembly step.
+
+
+### Adding pre-provisioned keys
+
+Mbed Crypto provides a way to register pre-existing secure element keys:
+`mbedtls_psa_register_se_key()`. This function adds the necessary metadata to
+persistent storage so that the secure element keys can be used by an
+application key ID.
+
+Provide which physical secure element slot the key is in with
+`psa_set_key_slot_number()`. This function operates on a key attributes
+structure. Complete the filling in of all necessary attributes and then call
+`mbedtls_psa_register_se_key()` to notify Mbed Crypto that the key exists,
+where the key exists, what format the key is, what the key can be used for, and
+so forth.
+
+
+### This example requires a certificate authority
+
+For the sake of this example, we'll assume you have three certificate
+authorities (CAs): a Root CA, a Server Sub-CA, and a Device Sub-CA. We'll
+assume you want to use openssl to run this test CA. We assume the following
+file names for the relevant files that comprise your CA, and that the file
+contents are PEM encoded.
+
+*Root CA*
+The Root CA is used to create subordinate CAs. Its certificate is used by both
+servers and devices in a certificate chain.
+- Private key in `RootCA.key`
+- Root certificate in `RootCA.crt`
+
+*Server Sub-CA*
+The Server Sub-CA is used to sign a server's CSR to produce a server
+certificate.
+- Private key in ServerSub.key
+- Server Sub-CA certificate in ServerSub.crt
+
+*Device Sub-CA*
+- Private key in DeviceSub.key
+- Device Sub-CA certificate in DeviceSub.crt
+
+We also assume the presence of the following certificate chains:
+
+*Server certificate chain*
+- The Root CA certificate followed by the Server Sub-CA certificate in
+  ServerChain.pem
+- The server certificate chain is used by devices to authenticate servers they
+  connect to.
+
+*Device certificate chain*
+- The Root CA certificate followed by the Device Sub-CA certificate in
+  DeviceChain.pem
+- The device certificate chain is used by servers to authenticate devices,
+  enabling mutually authenticated TLS connections.
+
+
+### Using device-generated CSRs
+
+This example generates certificate signing requests (CSRs) which can be used
+for TLS client authentication. The CSR is printed over the serial port, for
+use with your certificate authority (CA).
+
+The Device Sub-CA will consume the device-generated certificate signing request
+and produce a certificate for the device which you can use in the TLS with
+client authentication example also included in this repository (in the
+`mutual-tls` folder).
```

Comment:
Not found

Reply:
Not yet present. I can add this reference back in when the folder is present. Didn't want to add the mutual-tls example in the same PR, but I can add this reference from the provisioning example to the mutual-tls example in the mutual-tls example PR.